tests/src/test/scala/system/basic/WskActivationTests.scala

```diff
@@ -54,4 +57,27 @@ abstract class WskActivationTests extends TestHelpers with WskTestHelpers with J
       }, 10, Some(1.second))
     }
   }
+
+  it should "fetch result using activation result API" in withAssetCleaner(wskprops) { (wp, assetHelper) =>
+    val name = "hello"
+    val expectedResult = JsObject(
+      "result" -> JsObject("payload" -> "hello, undefined!".toJson),
+      "success" -> true.toJson,
+      "status" -> "success".toJson)
+
+    assetHelper.withCleaner(wsk.action, name) { (action, _) =>
+      action.create(name, Some(TestUtils.getTestActionFilename("hello.js")))
+    }
+
+    val run = wsk.action.invoke(name)
+
+    // Use withActivation() to reduce intermittent failures that may result from eventually consistent DBs
+    withActivation(wsk.activation, run) { activation =>
+      retry({
```

Comment:
isn't the retry here unnecessary? `withActivation` will have waited until there was an activation... can you add a comment that explains why the internal retry is necessary?